core/src/main/java/io/questdb/cairo/ColumnPurgeOperator.java

```diff
@@ -370,7 +370,11 @@ private boolean purge0(ColumnPurgeTask task, final ScoreboardUseMode scoreboardM
         } finally {
             if (scoreboardMode != ScoreboardUseMode.EXTERNAL) {
                 txnScoreboard = Misc.free(txnScoreboard);
-                Misc.free(txReader);
+                txReader = Misc.free(txReader);
+            } else {
+                // even though we take these things from the reader, we must not re-use them on the next run
+                txnScoreboard = null;
+                txReader = null;
```

Comment:
Looks like we may get an NPE due to this `null` assignment. `ColumnPurgeJob` calls both `purgeExclusive()` (`ScoreboardUseMode.EXCLUSIVE`) and `purge()` (`ScoreboardUseMode.INTERNAL`) methods. With this change both of them will have `txReader` set to `null` once they run first time. On the next call of any of these methods `setupScoreboard` will be `true`, so the `openScoreboardAndTxn()` method will be called. That method assumes presence of the `txReader` initialized in the operator's constructor, so it'll try making the following call:
```java
txReader.ofRO(path.$(), task.getPartitionBy());
```

This call will throw NPE.

Am I missing something?

Reply:
plot thickens

Reply:
fixed here: https://github.com/questdb/questdb/pull/5664/commits/1a13c6239d875f52ffa9b9b8a91dc516b1060675